c/driver_manager/adbc_driver_manager.cc

```diff
@@ -597,6 +628,7 @@ AdbcStatusCode AdbcStatementSetSqlQuery(struct AdbcStatement* statement,
 AdbcStatusCode AdbcStatementSetSubstraitPlan(struct AdbcStatement* statement,
                                              const uint8_t* plan, size_t length,
                                              struct AdbcError* error) {
+  AdbcErrorInit(error);
```

Comment:
Reset or something might be clearer? (Or is there a need for a function when it's only one line of code?)

Reply:
It is small - I can replace with a direct call to memset.

Reply:
After seeing @kou comments below I'll rename the fn to AdbcErrorReset and check for null as well.